rllib/policy/policy.py

```diff
@@ -534,6 +537,152 @@ def _create_exploration(self) -> Exploration:
             framework=getattr(self, "framework", "tf"))
         return exploration
 
+    def _get_default_view_requirements(self):
+        """Returns a default ViewRequirements dict.
+
+        Note: This is the base/maximum requirement dict, from which later
+        some requirements will be subtracted again automatically to streamline
+        data collection, batch creation, and data transfer.
+
+        Returns:
+            ViewReqDict: The default view requirements dict.
+        """
+
+        from ray.rllib.agents.dqn.dqn_tf_policy import PRIO_WEIGHTS
+
+        # Default view requirements (equal to those that we would use before
+        # the trajectory view API was introduced).
+        return {
+            SampleBatch.OBS: ViewRequirement(space=self.observation_space),
+            SampleBatch.NEXT_OBS: ViewRequirement(
+                data_col=SampleBatch.OBS,
+                shift=1,
+                space=self.observation_space),
+            SampleBatch.ACTIONS: ViewRequirement(space=self.action_space),
+            SampleBatch.PREV_ACTIONS: ViewRequirement(
+                data_col=SampleBatch.ACTIONS,
+                shift=-1,
+                space=self.action_space),
+            SampleBatch.REWARDS: ViewRequirement(),
+            SampleBatch.PREV_REWARDS: ViewRequirement(
+                data_col=SampleBatch.REWARDS, shift=-1),
+            SampleBatch.DONES: ViewRequirement(),
+            SampleBatch.INFOS: ViewRequirement(),
+            SampleBatch.EPS_ID: ViewRequirement(),
+            SampleBatch.AGENT_INDEX: ViewRequirement(),
+            SampleBatch.ACTION_DIST_INPUTS: ViewRequirement(),
+            SampleBatch.ACTION_LOGP: ViewRequirement(),
+            SampleBatch.VF_PREDS: ViewRequirement(),
```

Comment:
It's kind of odd to see agent-specific keys like VF_PREDS here. Can't we infer these dynamically always, and omit them from this dict?

Reply:
I can remove the extra-action fetch keys. These can indeed be added (if required) after the model's test call.
The others need to stay due to them being required (maybe) by the loss function. If any field is not required, it'll be removed automatically, so the user shouldn't really case. We always get the slimmest possible final view req dict.

Reply:
done